[Exception_Structure]

Ggyumalang opened the pull request:
## 작업 내용 (Content)
- 변경 사항 
  - ErrorCode ,ErrorResponse , BudException(CustomException) , GlobalExceptionHandler 작성 및 수정
  - 기존에 status 를 응답에 포함시키던 것을 없애고 errorCode와 message 만 반환하도록 수정

## Merge 전 필요 작업 (Checklist before merge)
- 이전 PR인 feat#1 pull 받아주세요!

## 희망 리뷰 완료 일 (Expected due date)
2023-04-04. Tue

## 리뷰 주셨으면 좋은 점
- PR을 올리고 나서 곰곰히 생각해보니 현재 제가 ErrorCode ,ErrorResponse , BudException(CustomException) 클래스들을 공통 모듈에 올려두었는데 이걸 공통으로 둘 게 아니라 각각의 모듈에 두어야할 것 같은데.. 여러분 생각은 어떠신 지 리뷰 주시면 감사하겠습니다.

## 리뷰 피드백 받은 후 수정 내용
 ErrorCode, ErrorResponse , BudException 3개의 클래스의 위치를 수정하였음.
 기존 core 모듈 -> user-api 모듈로 이동

thdefn committed:
chore : api 기본 응답 생성

개발에 필요한 공통 응답 엔티티 생성

Ggyumalang committed:
Merge branch 'main' of github.com:BuD2023/BuDBackend into feat#2

Ggyumalang committed:
Merge branch 'develop' of github.com:BuD2023/BuDBackend into feat#2

Ggyumalang committed:
Merge branch 'develop' of github.com:BuD2023/BuDBackend into feat#2

Ggyumalang committed:
feat : [Exception_Structure] Exception Handling의 기본 틀 구축

 ErrorCode, ErrorResponse , BudException, GlobalExceptionHandler 부분을 수정

thdefn commented on core/src/main/java/zerobase/bud/exception/BudException.java:
```diff
@@ -0,0 +1,25 @@
+
+    public BudException(ErrorCode errorCode) {
+        this.errorCode = errorCode;
+        this.message = errorCode.getDescription();
```
enum을 아주 효율적으로 활용하신 것 같습니다 ~!

Ggyumalang replied:
그 지금  에러코드 클래스를 core( 공통모듈 )에 넣어두었잖아요? 근데 각각의 모듈마다 errorCode가 나오는 게 다르고 많지 않을까???! 라는 생각이 들었었어요.. user-api(모듈1)와 나중에 만들어질 char-api(모듈2)는 errorcode가 다를 수 있을 것 같고 따로 관리하는 게 나을 수도 있을 것 같아서 user-api(모듈1) 밑으로 내리는 게 나을까?! 라는 생각을 좀 했습니다.

thdefn replied:
음 그럴수도 있을 것 같아요 ! 그러면 말씀하신대로 user-api 밑으로 내리는 것도 좋을 것 같습니다

thdefn reviewed (approved):
수고많으셨습니다 ! 그런데 각각의 모듈로 둔 다는 게 어떤 의미일까요 ~? core/.../dto 가 아니라 core/.../common/dto 이런 식을 말씀하시는걸까요 ?

xoals25 commented on core/src/main/java/zerobase/bud/type/ErrorCode.java:
```diff
@@ -0,0 +1,14 @@
+
+@Getter
+@AllArgsConstructor
+public enum ErrorCode {
```
너무 고생하셨습니다!
저는 ErrorCode 부분만 각 모듈에 만들면 더 좋을것 같은데 어떠신가요!?

Ggyumalang replied:
좋은 의견입니다!! 다른 모듈도 각각 Errorcode를 만들도록 하죠

xoals25 commented:
저는 공통 ErrorCode 따로 만들고 각각 모듈에 ErrorCode를 만들어서 사용하면 더 좋을것 같습니다!

Sfer7 reviewed (approved):
좋은 것 같아요! 수고하셨습니다~

Ggyumalang committed:
fix : [Revised_Exception_Structure] Exception Handling 의 기본 틀 수정

 ErrorCode, ErrorResponse , BudException 3개의 클래스의 위치를 수정하였음.
 기존 core 모듈 -> user-api 모듈로 이동
